okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -354,15 +345,15 @@ private void cancel(Status reason, boolean stopDelivery, Metadata trailers) {
         return;
       }
       cancelSent = true;
-      if (pendingData != null) {
+      if (canStart) {
         // stream is pending.
         transport.removePendingStream(OkHttpClientStream.this);
         // release holding data, so they can be GCed or returned to pool earlier.
         requestHeaders = null;
-        for (PendingData data : pendingData) {
-          data.buffer.clear();
-        }
-        pendingData = null;
+        pendingData.clear();
+        canStart = false;
+        flushPendingData = false;
```

Comment:
Clearing these two flags wasn't done in start(). I guess it doesn't matter if they are cleared, but it does seem misleading to do it one place and not the other. (So either remove the clearing here or add it to start()).

Reply:
removing those two from here since start may not be an entry point.